### PR TITLE
docs: architecture diagram and api docs

### PR DESCRIPTION
**Pull request by mohamed-cds**
Updated documentation and infrastructure diagram to wrap up this initial release.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Scan-files Infrastructure
+
+![AWS infrastructure diagram.  Full text description follows.](architecture.png)
+
+1. Virtual Private Cloud (VPC) with three public and three private subnets.
+1. API Gateway (AGW) to allow inbound/outbound communication, protected by Web Application Firewall (WAF), as main user request entry point and has an Lambda API as its target.
```
Comment (patheard) [inline docs/README.md:6]: Does the AGW allow outbound communication?  I thought the only outbound was from the Step Function lambda to AssemblyLine.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+1. NAT Gateways (3) to allow private subnets to communicate with AGW.
```
Comment (patheard) [inline docs/README.md:7]: Related to the above question, I didn't realize anything in the private subnets was talking back to the AGW.

**Review by patheard — COMMENTED** [review]
Just a few questions about the traffic flow of the API gateway and NAT gateways.

File: docs/README.md (hunk 1 above)
Comment (mohamed-cds) [reply docs/README.md]: reworded. Thanks!

File: docs/README.md (hunk 2 above)
Comment (mohamed-cds) [reply docs/README.md]: reworded as well

**Review by patheard — APPROVED** [review]
👍🏻

**Pull request merged**